jep/0000/README.adoc

```diff
@@ -0,0 +1,283 @@
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Not Submitted :information_source:
+
+| Type
+| Standards
+
+| Created
+| 2018-06-20
+//
+//
+// Uncomment if there is an associated placeholder JIRA issue.
+//| JIRA
+//| :bulb: https://issues.jenkins-ci.org/browse/JENKINS-nnnnn[JENKINS-nnnnn] :bulb:
+//
+//
+// Uncomment if there will be a BDFL delegate for this JEP.
+//| BDFL-Delegate
+//| :bulb: Link to github user page :bulb:
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| :bulb: Link to where discussion and final status announcement will occur :bulb:
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+
+== Abstract
+
+[TIP]
+====
+Give a short (200 word) description of the technical issue addressed.
+====
+
+Jenkins does not provide a homogeneous, easy to understand and easy use external API
+(through CLI and HTTP) to manage Jenkins data.
+
+Main contributions to this problem are: **lack of a real Resources API to be exposed** (raw model objects and sometimes
+just human readable text is returned nowadays), and **outdated and limited tooling** (CLI and HTTP APIs serialization
+is based on Stapler and Stapler annotations).
+
+This JEP is introducing both a new set of APIs to guide developers to write better external APIs and bringing
+new tooling to easily define the Resources layer to be exposed, as well as modern ways to build an HTTP/REST API.
+
+== Specification
+
+[TIP]
+====
+Provide a detailed specification what is being proposed.
+Be as technical and detailed as needed to allow new or existing Jenkins developers
+to reasonably understand the scope/impact of an implementation.
+====
+
+Main goals:
+
+. Isolate Model from public API: introduce a new _Resources layer_ based on new APIs (in advance the _Data API_)
+that can be leveraged for any data exposure (CLI and WS initially). It brings flexibility and improved
+API UX for CLI/WS users and developers.
+. Model can evolve whilst keeping backward compatibility in the Resources layer and Resources can
+transform, map or simplify the real model to adapt it for consumers.
+. The new _Data API_ is decoupled from Jenkins CLI and/or REST endpoints: the way to expose data is not a responsibility of the data itself.
+(However, this JEP also includes a proposal to expose data coming from the Data API through an HTTP API)
+. API messages are versioned (note the "version" attribute in messages) for easier evolution and correct handling of breaking changes.
```

Comment:
I haven't seen this `version` attribute anywhere. Can you put an example of it and how it is used as I don't understand it at the moment.

Reply:
Done.